lib/honeybadger/notice.rb

```diff
@@ -99,7 +99,7 @@ def url; @request[:url]; end
     attr_reader :local_variables
 
     # Public: The API key used to deliver this notice.
-    attr_reader :api_key
+    attr_accessor :api_key
```

Comment:
Is this change needed?

Reply:
Eventually I would like to refactor `Notice` so that all properties can be assigned to like this. I'm OK if we make this an `attr_accessor` now, but I'll still want to revisit `Notice` in the future. The main thing is that any sanitizing that we do (depth checking, applying params filters, etc.) must happen *after* all callbacks have run.

Reply:
I added this specifically for my use case that I outlined in #246. I noted the reason in the PR description as well.

Reply:
Sorry about that :sweat_smile: Sounds good